doc/cephadm/monitoring.rst

```diff
@@ -140,6 +140,88 @@ For example
 
           ceph config rm mgr mgr/cephadm/container_image_prometheus
 
+Using custom configuration files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By overriding cephadm templates, it is possible to completely customize the
+configuration files for monitoring services.
+
+Internally, cephadm already uses `Jinja2
+<https://jinja.palletsprojects.com/en/2.11.x/>`_ template to generate the
+configuration for all monitoring components. To be able to customize the
+configuration of Prometheus, Grafana or the Alertmanager it is possible to store
+a Jinja2 template that will be used for configuration generation instead. This
+template will be evaluated every time a service of that kind is deployed or
+reconfigured. That way the custom configuration is preserved and automatically
+applied on deployments of future services.
+
+.. note::
+
+  The configuration of the custom template is also preserved when the default
+  configuration of cephadm changes. If the updated configuration is to be used,
+  the custom template needs to be migrated *manually*.
+
+Option names
+""""""""""""
+
+The following templates for files that will be generated by cephadm can be
+overridden. These are the names to be used when storing with ``ceph config-key
+set``:
+
+- alertmanager_alertmanager.yml
+- grafana_ceph-dashboard.yml
+- grafana_grafana.ini
+- prometheus_prometheus.yml
+
+You can look up the file templates that are currently used by cephadm in
+`src/pybind/mgr/cephadm/templates`:
```

Comment:
I think single backticks for monospaced content are MarkDown, reStructuredText uses double backquotes.